torch/_inductor/lowering.py

```diff
@@ -5717,6 +5714,42 @@ def resize_storage_bytes_(variable, new_size):
     return variable
 
 
+@register_lowering(aten.resize, type_promotion_kind=None)
+def resize(x, size, *, memory_format=None):
+    x.realize()
+    dtype_item_size = get_dtype_size(x.get_dtype())
+    old_numel = V.graph.sizevars.evaluate_static_shape(sympy_product(x.get_size()))
+    new_numel = V.graph.sizevars.evaluate_static_shape(sympy_product(size))
+    if old_numel == new_numel:
+        return clone(x)
+
+    x = resize_storage_bytes_(x, new_numel * dtype_item_size)
```

Comment:
I don't think these semantics are correct. You resize the storage of `x` but return a new view of it so the tensor `x` itself is not updated. AFAICT the expected behavior is to return a new tensor with new storage.